tests/components/konnected/test_init.py

```diff
@@ -0,0 +1,452 @@
+"""Test Konnected setup process."""
+from unittest.mock import patch
+
+from homeassistant.components import konnected
+from homeassistant.components.konnected import config_flow
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockConfigEntry, mock_coro
+
+
+async def test_setup_with_no_config(hass):
+    """Test that we do not discover anything or try to set up a Konnected panel."""
+    with patch.object(hass, "config_entries") as mock_config_entries, patch.object(
+        konnected, "configured_devices", return_value=[]
+    ):
+        assert await async_setup_component(hass, konnected.DOMAIN, {}) is True
+
+    # No flows started
+    assert len(mock_config_entries.flow.mock_calls) == 0
+
+    # Default access token used
+    assert hass.data[konnected.DOMAIN][konnected.CONF_ACCESS_TOKEN] is not None
+    assert hass.data[konnected.DOMAIN][konnected.CONF_API_HOST] is None
+    assert konnected.YAML_CONFIGS not in hass.data[konnected.DOMAIN]
+
+
+async def test_setup_defined_hosts_known_auth(hass):
+    """Test we don't initiate a config entry if configured panel is known."""
+    with patch.object(hass, "config_entries") as mock_config_entries, patch.object(
+        konnected, "configured_devices", return_value=["aabbccddeeff", "112233445566"]
```

Comment:
Add mock config entries with corresponding data to hass instead of patching `configured_devices`.

Reply:
👍